### ISSUE TITLE
A question of "Randomness in Bowtie 2" in the manual

### ISSUE THREAD

**Issue by seahurt** [BenLangmead/bowtie2#290]
> The pseudo-random number generator is re-initialized for every read, and the seed used to initialize it is a function of the **read name**, nucleotide string, quality string, and the value specified with --seed.

If I have a read like:
```
@TPN12345:0001:ABCDEFG:1:12345:5678:6543 1:N:0:ATCGATCG
TTCCCNATTTACAGGTGCTGTACAGAGTAATGAGATGTGTGACGGCTGCAAACCAGGTGTTTTTTTCTGAGGCTG
+
AAAA/#EEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEAEEEEEEEE
```

Does "read name" means A (`TPN12345:0001:ABCDEFG:1:12345:5678:6543 1:N:0:ATCGATCG`) or B (`TPN12345:0001:ABCDEFG:1:12345:5678:6543`)?

As for my test, "read name" means A. 

I suggest to change to B. Because some program may add extra info to the header, which may change the original random number.

My bowtie2 version is: `version 2.3.5.1`

If anyone knows how to change the behavior , please let me know. Thanks so so much!

**Comment by BenLangmead** [comment]
You are correct that "name" means the whole name, not truncating at the first space.  It is common for FASTQ files to have read names with spaces (as you show), so this behavior matches what users will expect.

While SAM does not allow whitespace in read names, that truncation happens later within Bowtie 2, downstream from the read parsing.
